src/Context.tsx

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import { ApolloClient } from 'apollo-client';
+
+const { Provider, Consumer } = (React as any).createContext();
+
+export interface ApolloProviderProps<TCache> {
+  client: ApolloClient<TCache>;
+  children: React.ReactNode;
+}
+
+export const ApolloProvider: React.StatelessComponent<any> = ({
+  client,
+  children,
+}: ApolloProviderProps<any>) => <Provider value={client}>{children}</Provider>;
+
+export interface ApolloConsumerProps {
+  children: (client: ApolloClient<any>) => React.ReactElement<any> | null;
+}
+
+export const ApolloConsumer: React.StatelessComponent<ApolloConsumerProps> = Consumer;
```

Comment:
I think that there is still a use case for the old ApolloConsumer. It verified that a client had been set by the provider and if not it threw an error for the user. Should we keep the old `ApolloConsumer` and make it a wrapper around the consumer defined here?